### ISSUE TITLE
Discuss diel impact on the results

### ISSUE THREAD

**Issue by wkumler** [wkumler/MesoscopeMetabolomicsManuscript#8]
Reviewer 2 | 04 Oct 2024 | 12:20
Sampling:
The metabolome at different depths can be strongly affected by the sampling time. Day / night variations are substantial, mainly due to diel vertical migration but also to photosynthetic state of the phytoplankton. Discussing the results sampling times during the transect and during sampling of the HEE should be specified and statistical analysis on the impact (or lack of impact) on the data should be provided.

**Comment by wkumler** [comment]
Did the samples get prefiltered? Ask Rachel/Wei - DVM is much less of a concern for <53um population IIRC

**Comment by wkumler** [comment]
There's no detectable effect of diel things on most of the data. Fortunately this was pretty easy to test for the Falkor data (just adding time_class as a term in the formula) because the design is balanced and the highres eddy centers were all sampled at the same time each day so we don't need to worry about them either. The eddy transect is giving me problems though because the p-values seem to change wildly depending on the inclusion of the time_class term.

```
# A tibble: 4 × 5
  run                   param      `Surface  (15 m)` `Deep  (175 m)` `DCM  (110-120 m)`
  <chr>                 <chr>                  <dbl>           <dbl>              <dbl>
1 multivar_sla_only     sla_class             0.0881          0.0129             0.0022
2 multivar_time_only    time_class            0.0719          0.227              0.0272
3 multivar_sla_plustime sla_class             0.0066          0.0078             0.0026
4 multivar_sla_plustime time_class            0.006           0.120              0.0343
```

The results fortunately don't seem to differ all that much if `by` is set to `terms` instead of `margin` (and are exactly the same except for the p-values of SLA when time is included??)

```
# A tibble: 4 × 5
  run                   param      `Surface  (15 m)` `Deep  (175 m)` `DCM  (110-120 m)`
  <chr>                 <chr>                  <dbl>           <dbl>              <dbl>
1 multivar_sla_only     sla_class             0.0881          0.0129             0.0022
2 multivar_time_only    time_class            0.0719          0.227              0.0272
3 multivar_sla_plustime sla_class             0.054           0.0091             0.0011
4 multivar_sla_plustime time_class            0.006           0.120              0.0343
```

**Comment by wkumler** [comment]
When I double checked on a few specific compounds, I saw obvious changes in sucrose and trehalose but other compounds really didn't have much of a diel effect:

![image](https://github.com/user-attachments/assets/6d2ca2fd-b2f0-44a8-bbda-c968772358e9)

**Comment by wkumler** [comment]
I'm also worried that these effects are correlated and I don't have enough sample to really show a difference - for example, the highly-different Station 12 DCM happened to be sampled during the evening and that has way more biomass. I assume this is because of SLA things instead of time things because that station happens to be in the exact center of the gyre and because the other evening stations don't look as distinct at all.

```
# A tibble: 11 × 3
   station time_class sla_class  
   <chr>   <chr>      <chr>      
 1 4       Midday     Neither    
 2 5       Night      Anticyclone
 3 6       Midday     Anticyclone
 4 7       Midday     Anticyclone
 5 8       Evening    Anticyclone
 6 9       Night      Anticyclone
 7 10      Morning    Neither    
 8 11      Midday     Cyclone    
 9 12      Evening    Cyclone    
10 13      Night      Cyclone    
11 14      Morning    Anticyclone
```

**Comment by wkumler** [comment]
Then again, the variance *within* triplicates is much less than the variance in either depth OR time so maybe it makes sense that the PERMANOVA returns higher significance values when run with both together? So perhaps my initial analysis was erroneous by ignoring this factor. Also implies that perhaps a mixed-effects model would be most effective, with station as the random effect. I just don't know enough to understand how the blocking/nesting would work in a model like that.
![image](https://github.com/user-attachments/assets/7c09cced-0c6c-43fa-92ba-647fa050b066)

**Comment by wkumler** [comment]
Resolved with a9a9916

**Comment by wkumler** [comment]
Update: Anitra just pointed out that my times are probably wrong given Angie and Josh's results showing that trehalose and sucrose should be highest at the end of the day, not the beginning. Means that I've probably messed up time zones somewhere and need to recompile those.

**Comment by wkumler** [comment]
When I calculate local time (i.e. f521905) I get sampling times of midnight(ish) and noon(ish) for Falkor which I'm almost certain is incorrect:

```
  mutate(time=as.POSIXct(time, origin="1970-01-01", tz="UTC")) %>%
  mutate(local_time=as.POSIXct(time, tz="HST")) %>%
  mutate(time_class=c("Night", "Morning", "Midday", "Evening")[
    floor((as.numeric(format(local_time, "%H"))+3)/6)%%4+1
  ])
```

```
  station                time          local_time time_class
1      77 2018-04-07 22:57:24 2018-04-07 12:57:24     Midday
2      80 2018-04-08 11:14:18 2018-04-08 01:14:18      Night
3      62 2018-04-03 22:32:50 2018-04-03 12:32:50     Midday
4      64 2018-04-04 10:50:59 2018-04-04 00:50:59      Night
```

**Comment by wkumler** [comment]
Fixed with 0cfa931.
